src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
+          // So, we can just match the error and sent a less descriptive text
+          //
+          const char* invalidPolygon      = "Exterior shell of polygon is invalid";
+          const char* defaultErrorString  = "Internal mongo problem";
```

Comment:
I'd suggest this message:

"Error got querying MongoDB"

or similar (probably the above wording is not good English :)

I don't like "Internal mongo problem" two much due to: 1) "mongo" is ambiguous (it may refer to the mongo shell, which binary names the same, MongoDB is prefereable if we refer to the DB); 2) internal mongo problem could be interpreted as a problem in the MongoDB itself, which is not the case.